pkg/etcdenvvar/etcd_env.go

```diff
@@ -205,52 +207,36 @@ func getEtcdURLHost(envVarContext envVarContext) (map[string]string, error) {
 	return ret, nil
 }
 
-func getHeartbeatInterval(envVarContext envVarContext) (map[string]string, error) {
-	heartbeat := "100" // etcd default
-
-	infrastructure, err := envVarContext.infrastructureLister.Get("cluster")
+func getTuningProfileValues(envVarContext envVarContext) (map[string]string, error) {
+	etcd, err := envVarContext.etcdLister.Get("cluster")
 	if err != nil {
 		return nil, err
 	}
 
-	if status := infrastructure.Status.PlatformStatus; status != nil {
-		switch {
-		case status.Azure != nil:
-			heartbeat = "500"
-		case status.IBMCloud != nil:
-			if infrastructure.Status.PlatformStatus.IBMCloud.ProviderType == v1.IBMCloudProviderTypeVPC {
-				heartbeat = "500"
-			}
-		}
-	}
-
-	return map[string]string{
-		"ETCD_HEARTBEAT_INTERVAL": heartbeat,
-	}, nil
-}
-
-func getElectionTimeout(envVarContext envVarContext) (map[string]string, error) {
-	timeout := "1000" // etcd default
+	speed := etcd.Spec.HardwareSpeed
```

Comment:
How does this work by default when the `TechPreviewNoUpgrade` feature gate is not enabled?
Does this field exist and is just not set? Or do we error on trying to access this field?

If it's the latter then we probably need to check whether we have the feature gate enabled before we attempt to access this field.

Something similar to the following except we check for whether `TechPreviewNoUpgrade` is enabled on the FeatureGate status :
https://github.com/openshift/cluster-etcd-operator/blob/6c205c247a3965132d6c73d098126996f9768a95/pkg/operator/backupcontroller/backupcontroller.go#L80
https://github.com/openshift/cluster-etcd-operator/blob/6c205c247a3965132d6c73d098126996f9768a95/pkg/backuphelpers/backuphelpers.go#L10-L17

Reply:
I've added a check in the most recent update... thanks for finding this!